src/idpLoader.ts

```diff
@@ -0,0 +1,91 @@
+import * as https from "https";
+import { DOMParser } from "xmldom";
+import { log } from "./utils/logger";
+
+interface IDPMetadata {
+  cert: ReadonlyArray<string>;
+  entityID: string;
+  entryPoint: string;
+  logoutUrl: string;
+}
+
+const EntityDescriptorTAG = "md:EntityDescriptor";
+const X509CertificateTAG = "ds:X509Certificate";
+const SingleSignOnServiceTAG = "md:SingleSignOnService";
+const SingleLogoutServiceTAG = "md:SingleLogoutService";
+
+export async function parseIdpMetadata(): Promise<ReadonlyArray<IDPMetadata>> {
+  const IDP_METADATA_ULR =
+    process.env.IDP_METADATA_ULR ||
+    "https://registry.spid.gov.it/metadata/idp/spid-entities-idps.xml"; // TODO: remove default configuration?
```

Comment:
yes, make the server fail at startup when the value is not provided

Reply:
fix on commit d874196